receiver/kubeletstatsreceiver/internal/kubelet/accumulator.go

```diff
@@ -49,66 +49,47 @@ type metricDataAccumulator struct {
 	logger                *zap.Logger
 	metricGroupsToCollect map[MetricGroup]bool
 	time                  time.Time
+	mb                    *metadata.MetricsBuilder
 }
 
-const (
-	scopeName = "otelcol/kubeletstatsreceiver"
-)
-
 func (a *metricDataAccumulator) nodeStats(s stats.NodeStats) {
 	if !a.metricGroupsToCollect[NodeMetricGroup] {
 		return
 	}
 
-	md := pmetric.NewMetrics()
-	rm := md.ResourceMetrics().AppendEmpty()
-	fillNodeResource(rm.Resource(), s)
-
-	ilm := rm.ScopeMetrics().AppendEmpty()
-	ilm.Scope().SetName(scopeName)
-
-	startTime := pcommon.NewTimestampFromTime(s.StartTime.Time)
 	currentTime := pcommon.NewTimestampFromTime(a.time)
-	addCPUMetrics(ilm.Metrics(), metadata.NodeCPUMetrics, s.CPU, startTime, currentTime)
-	addMemoryMetrics(ilm.Metrics(), metadata.NodeMemoryMetrics, s.Memory, currentTime)
-	addFilesystemMetrics(ilm.Metrics(), metadata.NodeFilesystemMetrics, s.Fs, currentTime)
-	addNetworkMetrics(ilm.Metrics(), metadata.NodeNetworkMetrics, s.Network, startTime, currentTime)
+	addCPUMetrics(a.mb, metadata.NodeCPUMetrics, s.CPU, currentTime)
+	addMemoryMetrics(a.mb, metadata.NodeMemoryMetrics, s.Memory, currentTime)
+	addFilesystemMetrics(a.mb, metadata.NodeFilesystemMetrics, s.Fs, currentTime)
+	addNetworkMetrics(a.mb, metadata.NodeNetworkMetrics, s.Network, currentTime)
 	// todo s.Runtime.ImageFs
 
-	a.m = append(a.m, md)
+	a.m = append(a.m, a.mb.Emit(metadata.WithK8sNodeName(s.NodeName)))
```

Comment:
I don't think we need `a.m` anymore. metrics must be accumulated in the metrics builder instead. this line should be `a.mb.EmitForResource(metadata.WithK8sNodeName(s.NodeName)`

Reply:
The `MetricsData` function in `kubelet/metrics.go` returns a `[]pmetric.Metrics` but `mb.Emit()` returns a `pmetric.Metrics`.  If we use EmitForResource in each of the accumulator functions, how will we create the `[]pmetric.Metrics` in MetricsData using metrics builder?

Reply:
Why do we need to return `[]pmetric.Metrics` if all the data is accumulated in the metrics builder? See my latest comment in this section